components/bounty/edit-tag/controller.tsx

```diff
@@ -0,0 +1,47 @@
+import { useEffect } from "react";
+
+import { PROGRAMMING_LANGUAGES } from "assets/bounty-labels";
+
+import { useAppState } from "contexts/app-state";
+
+import BountyEditTagView from "./view";
+
+interface BountyEditTagControllerProps {
+  isEdit: boolean;
+  selectedTags: string[];
+  setSelectedTags: (v: string[]) => void;
+  preview?: boolean;
+}
+
+export default function BountyEditTagController({
+  selectedTags,
+  setSelectedTags,
+  isEdit = false,
+  preview = false,
+}: BountyEditTagControllerProps) {
+  const { state } = useAppState();
+
+  const TAGS_OPTIONS = PROGRAMMING_LANGUAGES.map(({ tag }) => ({
+    label: tag,
+    value: tag,
+  }));
+
+  function handleChangeTags(newTags) {
+    setSelectedTags(newTags.map(({ value }) => value));
+  }
+
+  useEffect(() => {
+    setSelectedTags(TAGS_OPTIONS.filter((tag) =>
```

Comment:
since bounty-data should be handed down by page-controller from get-server-side-props, we should be able to use a default-state for selectedTags and not need this useEffect